maths/prime_numbers.py

```diff
@@ -13,16 +13,16 @@ def primes(max: int) -> List[int]:
     >>> primes(1_000_000)[-1]
     999983
     """
-    max += 1
-    numbers = [False] * max
-    ret = []
-    for i in range(2, max):
-        if not numbers[i]:
-            for j in range(i, max, i):
-                numbers[j] = True
-            ret.append(i)
-    return ret
+    numbers: Generator = (i for i in range(1, (max + 1)))
+    for i in filter(lambda x: x > 1, numbers):
```

Comment:
__for i in [x for x in numbers if x > 1]:__

Reply:
filter returns an iterator, better then a full list, i guess

Reply:
__ for i in (x for x in numbers if x > 1):__